docs/databricks-monitoring/databricks-monitoring.md

```diff
@@ -0,0 +1,182 @@
+5. Under "Advanced Options", click on the "Spark" tab. Enter the following name-value pairs in the "Spark Config" text box:
+
+| Name | Value |
+|-------|--------|
+|spark.extraListeners|com.databricks.backend.daemon.driver.DBCEventLoggingListener,org.apache.spark.listeners.UnifiedSparkListener|
+|spark.unifiedListener.sink |org.apache.spark.listeners.sink.loganalytics.LogAnalyticsListenerSink|
+|spark.unifiedListener.logBlockUpdates|false|
+
+6. While still under the "Spark" tab, enter the following in the "Environment Variables" text box:
+* LOG_ANALYTICS_WORKSPACE_ID=[your Azure Log Analytics workspace ID](/azure/azure-monitor/platform/agent-windows#obtain-workspace-id-and-key)
+* LOG_ANALYTICS_WORKSPACE_KEY=[your Azure Log Analytics shared access signature](/azure/azure-monitor/platform/agent-windows#obtain-workspace-id-and-key)
+7. While still under the "Advanced Options" section, click on the "Init Scripts" tab. Go to the last line under the "Init Scripts section" Under the "destination" dropdown, select "DBFS". Enter "dbfs:/databricks/monitoring-staging/listeners.sh" in the text box. Click the "add" button.
+8. Click the "create cluster" button to create the cluster. Next, click on the "start" button to start the cluster.
+
+## Use the monitoring library in your code
+
+The Azare Databricks monitoring library supports the following types of metrics and message logging:
+
+### Apache Spark level events
+
+To send Apache Spark level event from Azure Databricks to your Azure Log Analytics workspace, follow these steps:
+
+1. Follow the instructions above to deploy the **spark-listeners-1.0-SNAPSHOT.jar** file that is built from the **spark-listeners** project.
+2. Once this is successfully deployed, metrics flow to your Azure Log Analytics workspace. You do not need to make any changes to your application code.
+
+### Apache Spark structured streaming sink for Azure Log Analytics
+
+To deploy the Apache Spark structured streaming sink for Azure Log Analytics in your Azure Databricks workspace, follow these steps:
+
+1. Follow the instructions above to deploy the **spark-listeners-loganalytics-1.0-SNAPSHOT.jar** file that is built from the **spark-listeners-loganalytics** project.
+2. Once this is successfully deployed, metrics flow to your Azure Log Analytics workspace. You do not need to make any changes to your application code.
+
+### Azure Databricks application metrics using Dropwizard
+
+To send application [metrics](https://spark.apache.org/docs/latest/monitoring.html#metrics) from your Azure Databricks application code to your Azure Log Analytics workspace using Dropwizard, follow these steps:
+
+1. Follow the instructions above to deploy the **spark-listeners-loganalytics-1.0-SNAPSHOT.jar** file that is built from the **spark-listeners-loganalytics** project.
+2. Create any [Dropwizard gauges or counters](https://metrics.dropwizard.io/4.0.0/manual/core.html) in your application code. 
```

Comment:
This is only applicable if the user wants to have custom metrics.  The built-in Spark metrics will automatically start flowing to the Azure Log Analytics workspace once Step 1 is complete.